addons/mail/static/src/models/attachment_image/attachment_image.js

```diff
@@ -4,24 +4,18 @@ import { registerNewModel } from '@mail/model/model_core';
 import { attr, many2one } from '@mail/model/model_field';
 import { clear, insert, insertAndReplace, replace } from '@mail/model/model_field_command';
 
-function factory(dependencies) {
-
-    class AttachmentImage extends dependencies['mail.model'] {
-
-        /**
-         * @override
-         */
+registerNewModel({
+    modelName: 'mail.attachment_image',
+    identifyingFields: ['attachmentList', 'attachment'],
+    lifecycle: {
```

Comment:
- I would just use `name` instead of `modelName`
- `lifecycle` => `lifecycles`

Reply:
The name `name` is already taken by Function and would conflict while generating models.